### PR TITLE
docs: add csv/historian/opentsdb

### PR DESCRIPTION
**Pull request by huolibo**

### CODE REVIEW

**Review by wingwing2005 — APPROVED** [review]

**Comment by wingwing2005** [comment]
@danielclow, please help to review the english part.

File: docs/en/04-data-in/01-ds/10-opentsdb.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+sidebar_label: OpenTSDB
+title: OpenTSDB Data Source
+description: This document describes how to extract data from OpenTSDB into a TDengine Cloud instance.
+---
+
+OpenTSDB data source is the process of writing data from the OpenTSDB server to the currently selected TDengine Cloud instance via a connection agent.
+
+## Prerequisites
+
+- Create an empty database to store your Kafka data. For more information, see [Database](../../../programming/model/#create-database).
```
Comment (danielclow) [inline docs/en/04-data-in/01-ds/10-opentsdb.md]: Kafka -> OpenTSDB

File: docs/en/04-data-in/01-ds/10-opentsdb.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- Ensure that the connection agent is running on a machine located on the same network as your OPC Server. For more information, see [Install Connection Agent](../install-agent/).
+
+## Procedure
+
+1. In TDengine Cloud, open **Data In** page. On the **Data Sources** tab, click **Add Data Source** button to open the new data source page. In the **Name** input, fill in the name of the data source and select the type of **OpenTSDB**, and in the **Agent** selection, select the agent you have already created, or if you have not created a agent, click the **Create New Agent** button to create it.
+2. In the **Target DB**, select the database of the current TDengine Cloud instance as the target database, since the time precision of data in OpenTSDB is millisecond, it is necessary to select a _`millisecond precision db`_ .
```
Comment (danielclow) [inline docs/en/04-data-in/01-ds/10-opentsdb.md:17]: I think item # 2 should be in "Prerequisites"

File: docs/en/04-data-in/01-ds/03-pi-system.md [hunk 3]
```diff
@@ -28,36 +28,17 @@ There is an additional charge for extracting PI System data. The charge depends
 3. In the **Name** field, enter a name for your data source.
 4. From the **Type** menu, select **PI**.
 5. From the **Agent** menu, select the connection agent for your PI System.
-   If you have not created a connection agent, see [Install Connection Agent](../install-agent/).
+   If you have not created a connection agent, see [Install Connection Agent](./00-install-agent).
```
Comment (danielclow) [inline docs/en/04-data-in/01-ds/03-pi-system.md]: Does this link work?

File: docs/en/04-data-in/01-ds/01-td.md [hunk 4]
```diff
@@ -14,12 +14,11 @@ The prerequisite for using this method is that the user needs to create the corr
...
-9. In the **Authentication** part, you can configure the user name and password for accessing the TDengine server on the remote edge side, the default value of the user name is root and the default value of the password is taosdata; if the data source is other instances of TDengine Cloud, you can select the token authentication method and configure the token of the instance.
-10. Click the **Connectivity Check** button to check whether the Cloud instance can be connected to the other TDengine Cloud instance service.
-11. After completing the above information, click the **Add** button to start the data synchronization from TDengine to TDengine Cloud instance directly.
+3. In the **Topic DSN** field, configure the DSN of the Topic that has been created in the TDengine of the data source.
+4. Click the **Connectivity Check** button to check whether the Cloud instance can be connected to the other TDengine Cloud instance service.
+5. in the **Subscribe Options** part, you can configure the subscription to start from the earliest data (earliest) or the latest data (latest), and the default is earliest.
```
Comment (danielclow) [inline docs/en/04-data-in/01-ds/01-td.md]: First word starts with capital letter ("In the ...") for # 5 and below

File: docs/en/04-data-in/01-ds/08-mqtt.md [hunk 5]
```diff
@@ -24,14 +24,16 @@ MQTT data source is the process of writing data from the MQTT server to the curr
...
-   - STable Name: used to configure the name of the sub-table, using the format of "prefix + &lcub;column type (target)&rcub;", for example, d&lcub;id&rcub;.
-   - Super Table Name: used to configure the super table name when synchronizing to TDengine;
-9. In the **Authentication** part, enter the user name and password of the MQTT connector when accessing the MQTT server. These two fields are optional, if you do not enter them, it means that anonymous authentication is used.
+7. You can click the **Connectivity Check** button to check whether the communication between the Cloud instance and the MQTT server is available.
+8. In the **MQTT Payload Parser** area, fill in the configuration parameters related to the payload parsing.
+   - In the **Message Body** field, fill in the sample data in the Kafka message body, for example: `{"id": 1, "message": "hello-word"}{"id": 2, "message": "hello-word"}`. This sample data will be used to configure the extraction and filtering conditions later. Click the **Preview** button to preview parse results.
```
Comment (danielclow) [inline docs/en/04-data-in/01-ds/08-mqtt.md]: Kafka -> MQTT

File: docs/en/04-data-in/01-ds/10-opentsdb.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+   - **Delay in seconds** is an integer ranging from 1 to 30, to migrate the out of order data, connector always waits for time specified here before reading them.
+
+6. **Advanced Options** is folded by default, and clicking on the right side can expand it.
+   - **Log Level** is used to configure the log level of the connector, supporting error, warn, info, debug, trace 5 levels, the default value is info.
+   - **Read Concurrency** is used to configure the maximum concurrency when reading data from the source OpenTSDB database, the default value is 50.
+   - **Write Concurrency** is used to configure the maximum concurrency when writing data to the target TDengine database, the default value is 50.
+   - **Batch Size** is used to configure the maximum batch size when writing data to the target TDengine database, the default value is 5000.
+7. After filling in the above information, click the **Submit** button to start the data synchronization from Kafka to TDengine.
```
Comment (danielclow) [inline docs/en/04-data-in/01-ds/10-opentsdb.md]: Kafka -> OpenTSDB

File: docs/en/04-data-in/01-ds/11-csv.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+   - **Skip the first N lines:**，Skip the first N lines for each CSV file.
+   - **Delimiter Char:** The field separator in a CSV line,The default value is `,`.
+   - **Quote Char:** The quote is used to enclose field values,The default value is `"`.
+   - **Comment Prefix:** If a line begins with the character given here, then that line will be ignored by the CSV parser,The default value is `#`.
+4. Upload a CSV file locally, such as test.csv, and then use the sample csv file to configure the extraction and filtering criteria.
+   - Click **Select File**, select test.csv, and click **Parse** to preview the identified columns, **Preview parsing results** on the right side of the screen.
+   - In the **Extract or Split From A column** field, fill in the fields to be extracted or split from the message body, for example: split the text field into `text_0` and `text_1` fields, select split Extractor, seperator fill in -, number fill in 2.Click the **Delete** button to delete the current extraction rule. Click the **Add** button to add more extraction rules. Click the **Preview** button to view the split result.
```
Comment (danielclow) [inline docs/en/04-data-in/01-ds/11-csv.md]: seperator -> separator

File: docs/en/04-data-in/01-ds/12-avevaHistorian.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+title: AVEVA Historian Data Source
+description: This document describes how to extract data from AVEVA Historian into a TDengine Cloud instance.
+---
+
+AVEVA Historian data source is the process of writing data from the AVEVA Historian server to the currently selected TDengine Cloud instance via a connection agent.
+
+## Prerequisites
+
+- Create an empty database to store your AVEVA Historian data. For more information, see [Database](../../../programming/model/#create-database).
+- Ensure that the connection agent is running on a machine located on the same network as your OPC Server. For more information, see [Install Connection Agent](../install-agent/).
```
Comment (danielclow) [inline docs/en/04-data-in/01-ds/12-avevaHistorian.md]: OPC Server -> AVEVA Historian instance?

Comment (huolibo) [reply docs/en/04-data-in/01-ds/12-avevaHistorian.md]: fixed

File: docs/en/04-data-in/01-ds/11-csv.md (hunk 7 above)
Comment (huolibo) [reply docs/en/04-data-in/01-ds/11-csv.md]: fixed

File: docs/en/04-data-in/01-ds/10-opentsdb.md (hunk 6 above)
Comment (huolibo) [reply docs/en/04-data-in/01-ds/10-opentsdb.md]: fixed

File: docs/en/04-data-in/01-ds/08-mqtt.md (hunk 5 above)
Comment (huolibo) [reply docs/en/04-data-in/01-ds/08-mqtt.md]: fixed

File: docs/en/04-data-in/01-ds/01-td.md (hunk 4 above)
Comment (huolibo) [reply docs/en/04-data-in/01-ds/01-td.md]: fixed

File: docs/en/04-data-in/01-ds/03-pi-system.md (hunk 3 above)
Comment (huolibo) [reply docs/en/04-data-in/01-ds/03-pi-system.md]: fixed

File: docs/en/04-data-in/01-ds/10-opentsdb.md (hunk 1 above)
Comment (huolibo) [reply docs/en/04-data-in/01-ds/10-opentsdb.md]: fixed

**Pull request merged**